src/senaite/core/upgrade/v02_03_000.py

```diff
@@ -420,3 +426,49 @@ def fixed_point_value_to_string(value, precision):
     str_value = template % (sign, front, fra)
     # strip off trailing zeros and possible dot
     return str_value.rstrip("0").rstrip(".")
+
+
+def setup_catalogs_order(portal):
+    """Ensures the order of catalogs portal types are bound to is correct
+    This is required because senaite.app.supermodel uses the first catalog
+    the portal type is associated with when retrieving brains
+    """
+    logger.info("Setup Catalogs order ...")
+
+    def sort_catalogs(id1, id2):
+        if id1 == id2:
+            return 0
+
+        # Catalogs sorted, senaite_* always first
+        senaite = map(lambda cat_id: cat_id.startswith("senaite_"), [id1, id2])
+        if not all(senaite) and any(senaite):
+            # Item starting with senaite always gets max priority
+            if id1.startswith("senaite_"):
+                return -1
+            return 1
```

Comment:
What do you think about a list of all catalogs, e.g. `[SETUP_CATALOG, SAMPLE_CATALOG ... AUDITLOG_CATALOG]` and use the index as sort key?

Reply:
I think is not necessary for now, keep in mind that other extension profiles might add new catalogs starting with `senaite_` prefix as well